GameEngine/src/States/IntroState.cpp

```diff
@@ -0,0 +1,190 @@
+#include "IntroState.hpp"
+
+IntroState::IntroState(StateMachine &t_machine, rtype::IRenderWindow *t_window,
+                       MusicPlayer &t_music_player, std::size_t t_flag,
+                       rtype::IGraphicLoader *t_graphic_loader, int *t_level,
+                       const std::string &t_path_to_sprite, bool t_replace,
+                       std::string t_ip, UdpClient *t_clientCom)
+    : State(t_machine, t_window, t_music_player, t_graphic_loader, t_level,
+            t_path_to_sprite, t_replace, t_ip, t_clientCom),
+      m_start_btn(Button(
+        "./assets/startBtn.png",
+        rtype::Vector2f{static_cast<float>(m_window->getSize().x / 2 + 135),
+                        static_cast<float>(m_window->getSize().y - 180)},
+        rtype::Vector2f{270, 130}, t_graphic_loader, false)),
+      m_flag(t_flag) {
+  m_bg_t = m_graphic_loader->loadTexture();
```

Comment:
Lager nochmal die SMFL Initialisierung in eine andere Funktion, dass der Konstruktor kleiner ist